lib/iris/tests/unit/experimental/regrid/test__CurvilinearRegridder.py

```diff
@@ -35,76 +40,152 @@
+        src_grid = self.src_grid
+        target_grid = self.tgt_grid
         regridder = Regridder(src_grid, target_grid, self.weights)
-        with mock.patch(self.func,
-                        return_value=mock.sentinel.regridded) as clr:
-            result = regridder(src_grid)
+        with mock.patch(self.func_setup,
+                        return_value=mock.sentinel.regrid_info) as patch_setup:
+            with mock.patch(
+                    self.func_operate,
+                    return_value=self.mock_slice_result) as patch_operate:
+                result = regridder(src_grid)
+        patch_setup.assert_called_once_with(
+            src_grid, self.weights, target_grid)
+        patch_operate.assert_called_once_with(
+            src_grid, mock.sentinel.regrid_info)
+        # The result is a re-merged version of the internal result, so it is
+        # therefore '==' but not the same object.
+        self.assertEqual(result, self.mock_slice_result)
 
-        clr.assert_called_once_with(src_grid, self.weights, target_grid)
-        self.assertIs(result, mock.sentinel.regridded)
+    def test_no_weights(self):
+        # Check we can use the regridder without weights.
+        src_grid = self.src_grid
+        target_grid = self.tgt_grid
+        regridder = Regridder(src_grid, target_grid)
+        # Note: for now, patch the internal partial result to a "real" cube,
+        # so we can do a cubelist merge, which is too complicated to mock out.
+        mock_slice_result = Cube([1])
+        with mock.patch(self.func_setup,
+                        return_value=mock.sentinel.regrid_info) as patch_setup:
+            with mock.patch(
+                    self.func_operate,
+                    return_value=self.mock_slice_result) as patch_operate:
+                result = regridder(src_grid)
+        patch_setup.assert_called_once_with(
+            src_grid, None, target_grid)
 
     def test_diff_src_from_init(self):
-        # Modify the names so we can tell them apart.
-        src_grid = self.ok.copy()
-        src_grid.rename('SRC_GRID')
-        target_grid = self.ok.copy()
-        target_grid.rename('TARGET_GRID')
+        # Check we can call the regridder with a different cube from the one we
+        # built it with.
+        src_grid = self.src_grid
+        target_grid = self.tgt_grid
         regridder = Regridder(src_grid, target_grid, self.weights)
-        src = self.ok.copy()
-        src.rename('SRC')
-        with mock.patch(self.func,
-                        return_value=mock.sentinel.regridded) as clr:
-            result = regridder(src)
+        # Note: for now, patch the internal partial result to a "real" cube,
+        # so we can do a cubelist merge, which is too complicated to mock out.
+        mock_slice_result = Cube([1])
+        # Provide a "different" cube for the actual regrid.
+        different_src_cube = self.src_grid.copy()
```

Comment:
Does just copying the source cube make it different enough from the original source cube to test this behaviour?